blockly/src/server/Server.java

```diff
@@ -108,38 +118,86 @@ public class Server {
     "WandUnten",
     "naheWand"
   };
-  private static final Stack<String> currently_repeating_scope = new Stack<>();
+  private final Stack<String> currently_repeating_scope = new Stack<>();
 
   /**
    * Object containing the variable HUD in the dungeon. This object is used to add new variables to
    * the HUD. It will add int variables and arrays to the HUD and update existing varaible values.
    */
-  public static VariableHUD variableHUD = null;
+  public VariableHUD variableHUD = null;
 
   /**
    * Constructor of the server. Sets the hero.
    *
    * @param hero The hero entity. Used to control the movement of the hero.
    */
-  public Server(Entity hero) {
-    Server.hero = hero;
+  private Server(Entity hero) {
+    this.hero = hero;
+  }
+
+  /**
+   * Singleton pattern. Get the instance of the server. If the server does not exist, create a new
+   * server object. The servers run on the {@link #DEFAULT_PORT}.
+   *
+   * @return Returns the server object.
+   * @throws MissingHeroException Throws a MissingHeroException if the hero entity could not be
+   *     found.
+   * @see #Server(Entity)
+   * @see #instances
+   * @see Game#hero()
+   */
+  public static Server Instance() {
```

Comment:
sind in Java Methoden nicht klein am Anfang?

Reply:
joa eigentlich schon, dachte das ist "schöner"/besser für Singleton Instance? Habe es aber umbenannt zu `instance`

Reply:
joa eigentlich schon, dachte das ist "schöner"/besser für Singleton Instance? Habe es aber umbenannt zu instance